code/scripting/api/objs/ai_helper.cpp

```diff
@@ -39,19 +41,65 @@ inline int aici_getset_helper(lua_State* L, float control_info::* value) {
 	return ade_set_args(L, "f", AI_ci.*value);
 }
 
-ADE_VIRTVAR(Pitch, l_AI_Helper, "number", "The pitch rate for the ship this frame, -1 to 1", "number", "The pitch rate, or 0 if the handle is invalid")
+inline int pi_rotation_getset_helper(lua_State* L, int axis) {
+	//Use with care! This works only if, as expected, the l_AI_Helper object is the one supplied to the luaai scripts this very frame
+	object_h ship;
+	float f = 0.0f;
+	if (!ade_get_args(L, "o|f", l_AI_Helper.Get(&ship), &f))
+		return ade_set_error(L, "f", 0.0f);
+
+	if (!ship.IsValid())
+		return ade_set_error(L, "f", 0.0f);
+
+	vec3d vel_limit;
+	// get the turn rate if we have Use_axial_turnrate_differences
+	if (The_mission.ai_profile->flags[AI::Profile_Flags::Use_axial_turnrate_differences]) {
+		vel_limit = Ship_info[Ships[ship.objp->instance].ship_info_index].max_rotvel;
+	}
+	else { // else get the turn time
+		float turn_time = Ship_info[Ships[ship.objp->instance].ship_info_index].srotation_time;
+
+		if (turn_time > 0.0f)
+		{
+			vel_limit.xyz.x = PI2 / turn_time;
+			vel_limit.xyz.y = PI2 / turn_time;
+			vel_limit.xyz.z = PI2 / turn_time;
+		}
+	}
+	vec3d acc_limit = ai_get_acc_limit(&vel_limit, ship.objp);
+
+	float currentThrustVel = ((ship.objp->phys_info.desired_rotvel.a1d[axis] - ship.objp->phys_info.rotvel.a1d[axis]) / AI_frametime) / acc_limit.a1d[axis];
```

Comment:
Should this not be
`((ship.objp->phys_info.desired_rotvel.a1d[axis] - ship.objp->phys_info.rotvel.a1d[axis]) * AI_frametime) / acc_limit.a1d[axis]`?

Reply:
In that line I don't calculate the new thrust, but rather the thrust as commanded by eventual preceeding AI calls to ``ai_turn_towards_vector``
So, if ``Δv = a * Δt``, then, where the acceleration ``a`` is proportional to our thrust, ``a = Δv/Δt``, no?

Reply:
Yeah, nevermind, that is correct. Although I think "Vel" here is probably the wrong term. I think perhaps "rate" like the doc uses is probably more accurate.